src/libraries/System.Private.CoreLib/src/System/Threading/Lock.cs

```diff
@@ -477,6 +477,8 @@ private ThreadId TryEnterSlow(int timeoutMs, ThreadId currentThreadId)
                     waitStartTimeTicks = Stopwatch.GetTimestamp();
                 }
 
+                using ThreadBlockingInfo.Scope threadBlockingScope = new(this, timeoutMs);
```

Comment:
At this point the initial spin-waiting would have already completed and it would shortly wait on the event. The spin-waiting done below is specific to the wait path, where a waiter when released also does a short spin-wait for the lock.